packages/backend/src/core/discovery/utils/diffDiscovery.ts

```diff
@@ -0,0 +1,67 @@
+import { EthereumAddress } from '@l2beat/types'
+
+import { DiscoveryContract } from '../DiscoveryConfig'
+import { ContractParameters } from '../types'
+import { diffContracts, FieldDiff } from './diffContracts'
+
+export interface DiscoveryDiff {
+  name: string
+  address: EthereumAddress
+  diff?: FieldDiff[]
+  type?: 'created' | 'deleted'
+}
+
+export function diffDiscovery(
+  committed: unknown[],
```

Comment:
Why is this `unknown[]`? ConfigReader should already return fully typed information